kie-server-parent/kie-server-services/kie-server-services-kafka/src/test/java/org/kie/server/services/jbpm/kafka/KakfaServerExtensionTest.java

```diff
@@ -0,0 +1,149 @@
+import org.apache.kafka.common.TopicPartition;
+import org.jbpm.bpmn2.core.Message;
+import org.jbpm.bpmn2.core.Signal;
+import org.jbpm.kie.services.impl.model.SignalDescImpl;
+import org.jbpm.services.api.DeploymentEvent;
+import org.jbpm.services.api.DeploymentService;
+import org.jbpm.services.api.ListenerSupport;
+import org.jbpm.services.api.ProcessService;
+import org.jbpm.services.api.model.DeployedUnit;
+import org.jbpm.services.api.model.ProcessDefinition;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.server.services.api.KieServerExtension;
+import org.kie.server.services.api.KieServerRegistry;
+import org.kie.server.services.impl.KieServerImpl;
+import org.mockito.Mockito;
+import org.mockito.verification.VerificationMode;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.Mockito.withSettings;
+
+public class KakfaServerExtensionTest {
+
+    private static class MockKafkaServerExtension extends KafkaServerExtension {
+
+        private MockConsumer<String, byte[]> consumer = new MockConsumer<String, byte[]>(OffsetResetStrategy.EARLIEST);
+
+        protected Consumer<String, byte[]> getKafkaConsumer() {
+            return consumer;
+        }
+
+        public MockConsumer<String, byte[]> getMockConsumer() {
+            return consumer;
+        }
+    }
+
+    private ProcessService processService;
+    private MockKafkaServerExtension extension;
+    private KieServerImpl server;
+    private KieServerRegistry registry;
+    private DeployedUnit deployedUnit;
+    private ProcessDefinition processDefinition;
+
+    @Before
+    public void setup() {
+        extension = new MockKafkaServerExtension();
+        server = mock(KieServerImpl.class);
+        registry = mock(KieServerRegistry.class);
+        KieServerExtension serverExtension = mock(KieServerExtension.class);
+
+        when(registry.getServerExtension(Mockito.anyString())).thenReturn(serverExtension);
+        ListenerSupport deployService = mock(ListenerSupport.class, withSettings().extraInterfaces(
+                DeploymentService.class));
+        processService = mock(ProcessService.class);
+        when(serverExtension.getServices()).thenReturn(Arrays.asList(deployService, processService));
+        extension.init(server, registry);
+        extension.serverStarted();
+        deployedUnit = mock(DeployedUnit.class);
+        processDefinition = mock(ProcessDefinition.class);
+        when(deployedUnit.getDeployedAssets()).thenReturn(Collections.singletonList(processDefinition));
+    }
+
+    @After
+    public void close() {
+        extension.destroy(server, registry);
+    }
+
+    @Test
+    public void testKafkaServerExecutorSignal() throws InterruptedException {
```

Comment:
Checked exception is not thrown within this method

```suggestion
    public void testKafkaServerExecutorSignal() {
```